app/controllers/concerns/foreman/controller/smart_proxy_auth.rb

```diff
@@ -81,11 +65,15 @@ def auth_smart_proxy(proxies = SmartProxy.unscoped.all, require_cert = true)
         logger.warn "No SSL cert supplied, falling back to reverse DNS for hostname lookup - request from #{request.remote_ip}"
         request_hosts = Resolv.new.getnames(request.remote_ip)
       end
-    elsif SETTINGS[:require_ssl]
-      logger.warn "SSL is required - request from #{request.remote_ip}"
-    else
+    elsif !SETTINGS[:require_ssl]
+      # SSL is explicitly disabled
       request_hosts = Resolv.new.getnames(request.remote_ip)
       request_hosts = [request.remote_ip] if request_hosts.empty?
+    else
+      # Request is not ssl request and ssl is not explicitly disabled.
+      # This should be unreachable branch and will keep request_hosts = nil thus return false
+      # just as a security measure
+      nil
```

Comment:
I think I prefer to not change it. It means we don't need to change the rubocop config. Also, it's still theoretically possible to reach it and then you would at least have a log line. IMHO there should be no changes to `auth_smart_proxy()`

Reply:
I disagree, as the unreachable branch should be lowest priority so last and do nothing as it's not expected to be executed. I'd have that as explicitly stated as possible. But it is a detail, so be it your way :)